tests/integration/test_gomod.py

```diff
@@ -170,6 +170,18 @@
             ),
             id="gomod_multiple_modules_missing_checksums",
         ),
+        # Test case checks if cachi2 can process go workspaces properly.
+        pytest.param(
+            utils.TestParameters(
+                repo="https://github.com/cachito-testing/cachi2-gomod.git",
+                ref="449f904ecf268856ef335b2033cb64cc417b7c26",
+                packages=({"path": "./workspace_modules/hello", "type": "gomod"},),
```

Comment:
I think we've already talked about this being the wrong interface
* the user shouldn't have to point to one of their workspace modules, they should be allowed to point to the workspace root
* cachi2 shouldn't treat one of the workspaces as special just because the user pointed to it

But if the version detection is not broken because of it, this is a good enough start

Reply:
> I think we've already talked about this being the wrong interface
> 
>     * the user shouldn't have to point to one of their workspace modules, they should be allowed to point to the workspace root
> 
>     * cachi2 shouldn't treat one of the workspaces as special just because the user pointed to it
> 
> 

Yeah, it was determined that from the CLI perspective nothing changes and the user is expected to point to the module they want to build (let's call it the "main" one) and so what you describe here requires a significant overhaul of the module, so based on that I accepted this as the working solution.